core/src/main/java/org/elasticsearch/action/bulk/BulkItemResponse.java

```diff
@@ -71,12 +83,77 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return builder;
     }
 
-    static final class Fields {
-        static final String _INDEX = "_index";
-        static final String _TYPE = "_type";
-        static final String _ID = "_id";
-        static final String STATUS = "status";
-        static final String ERROR = "error";
+    public static BulkItemResponse fromXContent(XContentParser parser, int id) throws IOException {
+        XContentParser.Token token = parser.nextToken();
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, token, parser::getTokenLocation);
+
+        token = parser.nextToken();
+        ensureExpectedToken(XContentParser.Token.FIELD_NAME, token, parser::getTokenLocation);
+
+        String currentFieldName = parser.currentName();
+        token = parser.nextToken();
+
+        final OpType opType = OpType.fromString(currentFieldName);
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, token, parser::getTokenLocation);
+
+        DocWriteResponse.DocWriteResponseBuilder builder = null;
+        CheckedConsumer<XContentParser, IOException> itemParser = null;
+        Supplier<? extends DocWriteResponse> itemSupplier = null;
```

Comment:
Nit: I think you can remove the itemSupplier and call builder.build() instead later.